### ISSUE TITLE
Can't start qtile on Debian

### ISSUE THREAD

**Issue by Ninofj** [qtile/qtile#1996]
# Issue description
I'm on Debian Buster and I installed qtile from source.  But when I try to log into qtile after seleting it on lightdm I get an error message 

**.xsessions-errors**
```
Xsession: unable to launch "python3 /home/nino/.local/bin/qtile start " X
session --- "python3 /home/nino/.local/bin/qtile start " not found; falling
back to default session.
```
My default is xfce

I checked multiple times. The binary for qtile is definitely in /home/nino/.local/bin/ as all my other pip3 programs.

I created a desktop file for qtile /usr/share/xsessions/qtile.desktop

```
[Desktop Entry] 
Name=Qtile
Comment=Qtile Session
Exec=python3 /home/nino/.local/bin/qtile start 
Type=Application 
Keywords=wm;tiling
```

# Qtile version
Version: 0.16.2

So what am I'm missing?

**Comment by tych0** [comment]
You have a string interpolation problem, it is trying to exec a binary called "python3 /home..." Instead of just "python3".

On Sun, Dec 6, 2020, at 8:48 AM, Fabian wrote:
> 

> Issue description

> I'm on Debian Buster and I installed qtile from source. But when I try to log into qtile after seleting it on lightdm I get an error message

> *.xsessions-errors*

> `Xsession: unable to launch "python3 /home/nino/.local/bin/qtile start " X
session --- "python3 /home/nino/.local/bin/qtile start " not found; falling
back to default session.
`
> My default is xfce

> I checked multiple times. The binary for qtile is definitely in /home/nino/.local/bin/ as all my other pip3 programs.

> I created a desktop file for qtile /usr/share/xsessions/qtile.desktop

> `[Desktop Entry] 
Name=Qtile
Comment=Qtile Session
Exec=python3 /home/nino/.local/bin/qtile start 
Type=Application 
Keywords=wm;tiling
`

**Comment by Ninofj** [comment]
Hey tych0, thanks for your answer. It works if I change it like this
```
[Desktop Entry] 
Name=Qtile
Comment=Qtile Session
Exec=/home/nino/.local/bin/qtile 
Type=Application 
Keywords=wm;tiling
```

But fails when adding start

```
[Desktop Entry] 
Name=Qtile
Comment=Qtile Session
Exec=/home/nino/.local/bin/qtile start
Type=Application 
Keywords=wm;tiling
```
It seems to work fine without "start". Is it necessary?

**Comment by tych0** [comment]
Yes, using `qtile` without start is deprecated. What DM are you using? Sounds like it's not adhering to the spec correctly: https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#exec-variables

**Comment by Ninofj** [comment]
I'm using lightdm

**Comment by ramnes** [comment]
What version of LigthDM are you using? It seems that this issue got fixed in version 1.0.4: https://github.com/canonical/lightdm/blob/02fa01c67289d5dfcd2287a6cd2c46090884ac69/debian/changelog#L1944

Anyway, the problem is not on our side so I'll just close this. Feel free to comment if you still have any issue! :)
